exporters/otlp/test/otlp_http_exporter_test.cc

```diff
@@ -11,12 +11,14 @@
 
 #  include "opentelemetry/exporters/otlp/protobuf_include_suffix.h"
 
+#  include "opentelemetry/ext/http/client/curl/http_client_curl.h"
```

Comment:
Sorry for responding late. The changes look good in general. I am just thinking if there is an alternate way of mocking the http_client library without directly referring to `ext::http::client::curl` in test code. Once we start supporting more HTTP libraries (#1146 & #1145), and if the libcurl is missing (say for windows if WinINet is used), the build for tests will fail with unresolved symbols for curl. As of now, we don't directly refer to http::client::curl in any of the exporters, but bring the right client through [HttpClientFactory](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/ext/include/opentelemetry/ext/http/client/http_client_factory.h). It would be good if we can continue the same way in tests too. Though I don't see any easy way of avoiding it ( maybe bringing a curl mock through HttpClientFactoryMock or something). Sorry if I am confusing here, let me think more and add more details.

Reply:
good point. so `HttpClientFactoryMock` which creates a `NosendHttpClient` with `MockSession` as its session and no inheritance from the curl client?

Reply:
Yes that's what I was thinking.